### ISSUE TITLE
Unit Upgrade Cost in data.json

### ISSUE THREAD

**Issue by denniske** [SiegeEngineers/aoe2techtree#70]
Hi, I want to display the upgrade cost and research time in my app. Is it possible to add this to your data file?

I know I can use the description text e.g.

> Upgrades: range, attack (University); to Heavy Scorpion 1000F, 1100W (Siege Workshop); more resistant to Monks (Monastery).

But there the research time is missing and I am displaying the range/attack upgrades etc. in another way already.

**Comment by HSZemi** [comment]
Oh sorry, I thought there was already an answer here.

In order to do that, one would essentially have to map the upgrade techs to the units. That could be semi-automated by looking at the upgrade Effects and which units get upgraded to which, but since for example the upgrade Effect for Champion contains upgrade commands for the whole Militia line, that data would have to get cleaned up afterwards.

**Comment by denniske** [comment]
Yeah that would be cool. I am parsing the description at the moment which is not optimal:
https://github.com/denniske/aoe2companion/blob/master/data/src/helper/units.ts#L2823 (getUnitUpgradeCost)

**Comment by denniske** [comment]
I have seen that you have added unit cost to aoe2techtree and the following seem to have been switched up with huns atheism tech:

demolition ship 
fire ship

![image](https://user-images.githubusercontent.com/1076211/143298427-bb942687-17e9-4c83-b792-6a08b66b5a33.png)


![image](https://user-images.githubusercontent.com/1076211/143286156-753bbb84-d4fa-4907-9d20-cd582b2dca48.png)

**Comment by HSZemi** [comment]
Funny. Demolition Ship and Fire Ship actually have `"Trigger Tech ID":21,`, which is the **unit id** of War Galley, not the ID of the War Galley upgrade. I'll have to check if I can deduce such cases automatically or if I have to hardcode something for that case. Thanks for the report!

**Comment by HSZemi** [comment]
Hardcoded for now :shrug:
